chain/epoch_manager/src/lib.rs

```diff
@@ -497,13 +503,13 @@ impl EpochManager {
             Ok(next_next_epoch_info) => next_next_epoch_info,
             Err(EpochError::ThresholdError { stake_sum, num_seats }) => {
                 warn!(target: "epoch_manager", "Not enough stake for required number of seats (all validators tried to unstake?): amount = {} for {}", stake_sum, num_seats);
-                let mut epoch_info = next_epoch_info.clone();
+                let mut epoch_info = (*next_epoch_info).clone();
```

Comment:
`next_epoch_info` is already an `Arc` AFAICT, so `Arc::clone` here, and pull `Arc::new` into the relevant branches.

(And more generally, it is pretty hard to tell which `Clone` impl is being invoked here, and it sort of matters, so prudent to use `T::clone(...)` to invoke, over a method call)

Reply:
>AFAICT, so Arc::clone here, and pull Arc::new into the relevant branches.

Hm, I think this doesn't work (or I am misunderstanding the suggestion). Here, we want and independent copy to mutate it. Ie, even `Arc::make_mut` doesn't seem to be what we need here. 

> so prudent to use T::clone(...) to invoke, over a method call

Thanks, I'll add this to my idiom store!